tests/unit/test_tracing.c

```diff
@@ -1377,37 +1378,165 @@ SENTRY_TEST(set_trace_id_with_txn)
         = "2674eb52d5874b13b560236d6c79ce8a-a0f9fdf04f1a63df-1";
     const char *txn_trace_id = "2674eb52d5874b13b560236d6c79ce8a";
     const char *txn_parent_span_id = "a0f9fdf04f1a63df";
+    sentry_transaction_context_t *tx_ctx_scoped
+        = sentry_transaction_context_new("wow!", NULL);
+    sentry_transaction_context_update_from_header(
+        tx_ctx_scoped, "sentry-trace", trace_header);
```

Comment:
We should also test doing this after `set_trace` was called; IMO it should get priority over the value gotten from `set_trace`.

Reply:
Yes, iirc that was the idea (i.e., explicit traces from external contexts overriding the inherited trace) from our last sync on the topic. Let's add any test that raises similar questions until we hit an implementation blocker.

Reply:
I mean, just to be clear: I am also not quite sure what it means if a user introduces an external trace that overwrites the traces introduced by SDKs that run "above" the Native SDK. And I mean this concretely for the Native SDK and generally. The latter seems to be answered by "Sentry does it that way", but for the Native SDK specifically, it also highlights the fact that we only have a single scope. The above behavior makes perfect sense if a thread continues an external trace, while all others stay inside the propagated trace.